lib/safe-pg-migrations/plugins/idem_potent_statements.rb

```diff
@@ -26,6 +26,10 @@ def remove_index(table_name, options = {})
       SafePgMigrations.say("/!\\ Index '#{index_name}' not found on table '#{table_name}'. Skipping statement.", true)
     end
 
+    def add_foreign_key(from_table, to_table, **options)
+      return super unless foreign_key_exists?(from_table, to_table)
```

Comment:
And just thought about it, but in the `add_belongs_to` test case below, if one the operation fails after the actual column creation, we can not replay the migration, as `add_column` is not idempotent.

Reply:
Created a TT for that 👍  https://doctolib.atlassian.net/browse/TT-2094